### PR TITLE
User wins when they select all of the empty tiles

### PR DESCRIPTION
**Pull request by lottetreg**
Checks if all the empty tiles on a board have been selected. I added another overloaded `Game.play()/1` function to handle game states with a status of `:player_won`.

Other updates:

- Created a `test/integration` directory and moved the contents of `GameTest` into there. `GameTest` is the highest level module (except for `Minesweeper` that kicks off the whole thing), so I just have integration tests instead of unit tests for it
- Moved the `MockRandomizer` set up into a helper module and included it in all tests
- Added unit tests for the `Move` module

### CODE REVIEW

File: test/game_rules_test.exs [hunk 1]
```diff
@@ -5,25 +5,41 @@ defmodule GameRulesTest do
 
   setup :verify_on_exit!
 
-  test "a game is over if a bomb has been selected" do
+  test "the player has lost if a bomb has been selected" do
```
Comment (Thomascountz) [inline test/game_rules_test.exs:8]: I know we suggested removing the duplication caused by having to explicitly define what the `MockRandomizer` should return, but now I think this test is a bit hard to read. What do you think? Is there a happy medium?

File: test/integration/player_loses_when_they_land_on_a_bomb_test.exs [hunk 2]
```diff
@@ -1,30 +1,22 @@
-defmodule GameTest do
+defmodule PlayerLosesWhenTheyLandOnABombTest do
```
Comment (Thomascountz) [inline test/integration/player_loses_when_they_land_on_a_bomb_test.exs:1]: I like the idea of calling these integration tests :D

**Review by Thomascountz — APPROVED** [review]
LGTM! I just had one non-blocking question about your test setup 💃

**Comment by lottetreg** [comment]
Thank you, @Thomascountz! I took your advice and made the `MockRandomizer` parts easier to read here: https://github.com/lottetreg/minesweeper/pull/15

**Comment by lottetreg** [comment]
Thank you, @Thomascountz! I took your advice and made the `MockRandomizer` parts easier to read here: https://github.com/lottetreg/minesweeper/pull/15

**Review by ecmendenhall — APPROVED** [review]

**Pull request merged**